101-bundle-json.md

```diff
@@ -442,16 +442,18 @@ Implementations MAY support user-defined additional actions as well. Such action
 
 ```json
 "actions": {
-    "status":{
+    "io.cnab.status":{
         "modifies": false
     },
-    "migrate":{
+    "io.cnab.migrate":{
         "modifies": false
     }
 }
 ```
 
-The above declares to actions: `status` and `migrate`. This means that the associated invocation images can handle requests for `status` and `migrate` in addition to `install`, `upgrade`, and `uninstall`.
+The action _name_ SHOULD be namespaced by the CNAB implementation which generated the bundle. 
```

Comment:
This presumes a close relationship between the code which generates the bundle and the runtime which consumes it. I think the namespace should actually be that of the runtime which will recognise the action rather than the code which generates the bundle.

Reply:
As I read it, the suggestion is that the "bundle generator" is the thing that is responsible for adding the namespaced ID, and does not say anything about how that operates. I think Silvin is merely expressing what agent is responsible for _writing the namespaced data_.

Reply:
I think the "bundle generator" is more aware of what actions it put into the invocation image and how it should be run, more than the "bundle runtime".

Reply:
I think if you remove "by the CNAB implementation which generated the bundle." the meaning of this doesn't change and the spec has less words in it. I think the meaning of "CNAB implementation" is up for debate and removing it removes the potential for ambiguity.

Reply:
Ok, let me remove this part then 👍